gtsfm/loader/colmap_loader.py

```diff
@@ -0,0 +1,179 @@
+               e.g. for 1080p (1920 x 1080), max_resolution would be 1080
+        """
+        self._use_gt_intrinsics = use_gt_intrinsics
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._max_frame_lookahead = max_frame_lookahead
+        self._max_resolution = max_resolution
+        #import pdb; pdb.set_trace()
+
+        self._wTi_list, img_fnames = io_utils.read_images_txt(fpath=os.path.join(colmap_files_dirpath, "images.txt"))
+        self._calibrations = io_utils.read_cameras_txt(fpath=os.path.join(colmap_files_dirpath,"cameras.txt"))
+        if len(self._calibrations) == 1:
+            # shared calibration!
+            import pdb; pdb.set_trace()
+            self._calibrations = [self._calibrations] * len(img_fnames)
+
+        if self._calibrations is None or len(img_fnames) != len(self._calibrations):
+            self._use_gt_intrinsics = False
+
+        # import pdb; pdb.set_trace()
+        # self._wTi_list = [self._wTi_list[i] for i in [0, 8]]
+        # img_fnames = [img_fnames[i] for i in [0,8]] 
+        # self._calibrations = [self._calibrations[i] for i in [0,8]]
+
+        # preserve COLMAP ordering of images
+        self._image_paths = []
+        for img_fname in img_fnames:
+            img_fpath = os.path.join(images_dir, img_fname)
+            from pathlib import Path
+            if not Path(img_fpath).exists():
+                continue
+            self._image_paths.append(img_fpath)
+
+
+        self._num_imgs = len(self._image_paths)
+        print(f"Has {self._num_imgs} images")
+        # self.get_camera_intrinsics(0)
+
+        # import pdb; pdb.set_trace()
+        # # read one image, to check if we need to downsample the images
+        # img = io_utils.load_image(self._image_paths[0])
+
+        # sample_h, sample_w = img.height, img.width
+        # if min(sample_h, sample_w) > self._max_resolution:
+        #     self._downsample_factor, self.target_h, self.target_w = img_utils.get_exact_downsample_factor(img, self._max_resolution)
+        # else:
+        #     self._downsample_factor = 1
+        self._downsample_factor = 4
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            the number of images.
+        """
+        return self._num_imgs
+
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index.
+        """
+
+        if index < 0 or index > self.__len__():
+            raise IndexError("Image index is invalid")
+
+        img = io_utils.load_image(self._image_paths[index])
+        img = img_utils.resize_image(img, new_height=img.height//self._downsample_factor, new_width=img.width//self._downsample_factor)
```

Comment:
If this usage is common, I’d have the utils accept the scale as an argument.

Reply:
Yep. Should go in utils.

Reply:
Just to be clear -- we don't have a utils function for this at the present. We have to be a little explicit here since the requested aspect ratio may not be possible.

So we may want different scaling factors for focal length for each axis.